src/sssom/validators.py

```diff
@@ -23,14 +23,37 @@ def validate(msdf: MappingSetDataFrame, validation_types: List[SchemaValidationT
         VALIDATION_METHODS[vt](msdf)
 
 
+def print_linkml_report(report: ValidationReport, fail_on_error: bool = True):
+    validation_errors = 0
+
+    if not report.results:
+        logging.info('The instance is valid!')
+    else:
+        for result in report.results:
+            validation_errors += 1
+            if result.severity == Severity.ERROR:
+                logging.error(result.message)
+
+    if fail_on_error and validation_errors:
+        raise ValidationError(f"You mapping set has {validation_errors} validation errors!")
+
 def validate_json_schema(msdf: MappingSetDataFrame) -> None:
     """Validate JSON Schema using linkml's JsonSchemaDataValidator.
 
     :param msdf: MappingSetDataFrame to eb validated.
     """
-    validator = ReferenceValidator(SchemaView(SCHEMA_YAML))
+    validator = Validator(
+        schema=SCHEMA_YAML,
+        validation_plugins=[
+            JsonschemaValidationPlugin(closed=True),
```

Comment:
Yes, the `is not a valid URI or CURIE` error was previously emanating from [Python code](https://github.com/linkml/linkml-runtime/blob/c7815cb81c539f919e2ec48d2ca38d06da5aeb18/linkml_runtime/utils/metamodelcore.py#L103-L104) in `linkml-runtime` that is invoked by the generated `Mapping` dataclass.

JSON Schema does sort of have a concept of URIs via the `format` [keyword](https://json-schema.org/understanding-json-schema/reference/string#resource-identifiers), but LinkML's JSON Schema generator doesn't currently take advantage of that. We could take a look at using that. Alternatively we could look into implementing URI format checking as a separate validation plugin.

Reply:
Thanks. If it worked, would the `PydanticValidationPlugin` do this somehow or not? Type checking for entity references seems very fundamental to all we do!

Reply:
It would be up to the Pydantic generator to do that, not the `PydanticValidationPlugin`. I'm less familiar with that generator since it was majorly refactored recently, but I don't _believe_ it does that.

Reply:
https://github.com/linkml/linkml/issues/2116